### PR TITLE
Tidy code a bit

### PR DESCRIPTION
**Pull request by Sorien**

### CODE REVIEW

**Comment by Sorien** [comment]
i'm done .. it's based on idea/phpstorn code inspection  ...  there should be no function code changes .. hf with review :) ...

**Comment by Sorien** [comment]
Travis error is unrelated

**Comment by azokal** [comment]
I just finish to review all, I will speak about that with @ambroisemaupate when he come back. Thanks :)

File: src/Roadiz/Utils/TwigExtensions/BlockRenderExtension.php [hunk 1]
```diff
@@ -93,6 +94,8 @@ public function blockRender(NodesSources $nodeSource, $themeName = "DefaultTheme
             } else {
                 throw new \Exception("Invalid name formatting for your theme.", 1);
             }
+        } else {
+            throw new \Exception("Invalid NodesSources.", 1);
```
Comment (ambroisemaupate) [inline src/Roadiz/Utils/TwigExtensions/BlockRenderExtension.php]: :+1:

File: src/Roadiz/Utils/XlsxExporter.php [hunk 2]
```diff
@@ -86,7 +86,7 @@ public static function exportXlsx($data, $keys = [])
 
         $headerkeys = $keys;
 
-        foreach ($data as $key => $answer) {
+        foreach ($data as $answer) {
```
Comment (ambroisemaupate) [inline src/Roadiz/Utils/XlsxExporter.php]: :+1:

File: src/Roadiz/Utils/Document/DownscaleImageManager.php [hunk 3]
```diff
@@ -201,7 +202,7 @@ protected function createDocumentFromImage(Document $originalDocument, Image $pr
              * If raw document size is still outiside new maxSize cap
              * We keep intact raw document, and we just update downscaled doc.
              */
-            $rawDocument = $originalDocument->getRawDocument();
+            $originalDocument->getRawDocument();
```
Comment (ambroisemaupate) [inline src/Roadiz/Utils/Document/DownscaleImageManager.php]: Ok, the `$rawDocument` var is useless, you can even delete the entire line.

File: src/Roadiz/Core/Entities/Document.php [hunk 4]
```diff
@@ -435,7 +448,8 @@ public function getFolders()
     }
 
     /**
-     * @param Document $folder
+     * @param Document|Folder $folder
```
Comment (ambroisemaupate) [inline src/Roadiz/Core/Entities/Document.php]: `Document` type hinting is wrong. You can leave only `Folder` one.

File: src/Roadiz/CMS/Importers/GroupsImporter.php [hunk 5]
```diff
@@ -59,8 +59,8 @@ public static function importJsonFile($serializedData, EntityManager $em)
                     /*
                      * then persist each role
                      */
-                    $role = $em->getRepository('RZ\Roadiz\Core\Entities\Role')
-                               ->findOneByName($role->getName());
+                    $em->getRepository('RZ\Roadiz\Core\Entities\Role')
+                       ->findOneByName($role->getName());
```
Comment (ambroisemaupate) [inline src/Roadiz/CMS/Importers/GroupsImporter.php]: This foreach loop is not needed anymore.

**Comment by ambroisemaupate** [comment]
@Sorien Thanks a lot for this huge pull request. I’ve just commented some dead code that appear following your fixes. I’ll wait for your final fixes before merging your PR.

**Comment by Sorien** [comment]
i have bunch of other fixes ready to commit

**Comment by ambroisemaupate** [comment]
@Sorien Welcome to roadiz official contributors! :smile:

**Pull request merged**